ga4gh/datamodel/variants.py

```diff
@@ -477,6 +477,22 @@ def getNumVariants(self):
         # TODO How do we get the number of records in a VariantFile?
         return 0
 
+    def _updateCallSetIds(self, variantFile):
+        """
+        Updates the call set IDs based on the specified variant file.
+        """
+        checkConsistency = True
+        if self._callSetIds is None:
+            self._callSetIds = []
+            checkConsistency = False
+
+        for sample in variantFile.header.samples:
+            if checkConsistency:
+                if sample not in self._callSetIds:
+                    raise Exception("Inconsistent sample names in VCF")
+            else:
+                self._callSetIds.append(sample)
```

Comment:
Is it possible for the same `sample` to be encountered more than once when `checkConsistency` is `False` (i.e., when `self._callSetIds` is being built up within this method)? If so, is adding it again the desired behavior?